### ISSUE TITLE
Fix S3168 FN: Support virtual interface methods

### ISSUE THREAD

**Issue by gregory-paidis-sonarsource** [SonarSource/sonar-dotnet#6305]
### Description

FN for S3168 (AsyncVoidMethod).

Using C#11 virtual interface methods, the rule is not raising an issue.

### Repro steps

```csharp
public interface ISomeInterface
{
    static virtual async void SomeVirtualMethod() // FN
    {
        return;
    }
}
```

### Expected behavior

The rule should raise an issue.

### Actual behavior

The rule does not raise an issue.
